frontend/src/components/application/AppVendingControls/OwnershipTokens/TokenCancelButton.tsx

```diff
@@ -20,43 +20,41 @@ const TokenCancelButton: FunctionComponent<Props> = ({
 }) => {
   const { t } = useTranslation()
 
-  const {
-    execute: onClick,
-    value,
-    status,
-    error,
-  } = useAsync(
-    useCallback(
-      () => cancelVendingTokens(appId, [token.token]),
-      [appId, token.token],
-    ),
-    false,
-  )
-
-  useEffect(() => {
-    if (error) {
-      toast.error(t(error))
-    }
-  }, [t, error])
-
-  useEffect(() => {
-    if (value?.[0].status === "cancelled") {
-      setState("cancelled")
-      toast.success(t("token-cancelled"))
-    }
-  }, [value, setState, t])
-
-  if (status === "pending") {
+  const cancelVendingTokens = useMutation({
+    mutationFn: () => {
+      return vendingApi.cancelTokensVendingappAppIdTokensCancelPost(
+        appId,
+        [token.id],
+        {
+          withCredentials: true,
+        },
+      )
+    },
+    onSuccess: (data) => {
+      if (data.data[0].status === "cancelled") {
+        setState("cancelled")
+        toast.success(t("token-cancelled"))
+      }
+    },
+    onError: (error) => {
+      toast.error(t(error as string))
+    },
```

Comment:
Will give this PR a test tomorrow morning and just merge if it's good :+1:

General note: I believe `onSuccess` and `onError` are deprecated. Same effects can be achieved inside the query function.

Reply:
Yes, they are deprecated, but I wasn't sure if that applies to mutations too. But it probably does.

Reply:
Also, we might want to address the `DEFAULT_SETUP` problem first.

Reply:
Then again, this isn't used, so creating an issue for that would also be fine in my book.